OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -226,7 +227,7 @@ protected override void OnFirstRun(Actor self)
 			if (target.IsValidFor(self))
 			{
 				QueueChild(new Fly(self, target, target.CenterPosition));
-				QueueChild(new Fly(self, target, exitRange, WDist.MaxValue, target.CenterPosition));
+				QueueChild(new FlyForward(self, exitRange));
```

Comment:
I'm not sure this will do for moving targets. The aircraft won't achieve the proper distance if the target is moving in the same direction as the attacker. This might be especially relevant for air-to-air attacks, but is still a slight regression for the current first-party use-cases (mig and yak). 

This is not an issue for the explicit strafe case, because there it is targeting a fixed spot.

I also don't think the new behaviour is necessarily that much more sensible as default behaviour for units that can slide but can't hover (units that can hover do not make use of this code path anyway). It is still a weird oscillating back-and-forth that isn't optimal given it's flight characteristics and weapon range. The most sensible default behaviour in this case would be a circle strafe, maintaining a fixed distance and constantly facing towards the target.

So I think this is a good change for the Strafe attack type, when you explicitly want this particular flight pattern, but not for the default.

Reply:
Hm good point. I hadn't considered moving targets, so I'll have to think about this again. Are you sure about the oscillating behaviour though? (IIrc I set the MiG testcase up for this branch, but don't remember.)

Reply:
The behaviour for the MiG here is as follows:
- It first moves towards the target, facing it&mdash;so far so good.
- It then moves away from the target in a straight line, facing _away_ from the target so it cannot fire even though it is able to slide in any direction and should just be facing the target. 
- As soon as it reaches the exitRange it is immediately jerked back towards the target&mdash;still facing away from the target!&mdash;slowly turning and only facing the target when it is already almost on top of the target.

Steps 2 and 3 then repeat ad infinitum, meaning that in a sustained attack (or an attack started while already within weapon range) the plane is actually unable to fire it's weapon during most of the attack run while simultaneously being always closer than it needs to be thereby unnecessarily exposing itself.

So it is actually strictly worse than the behaviour shown in the gif, since in that case it is at least always facing the target and neither of the maneuvers are impossible given the defined flight characteristics.

As I said, the most logical behaviour would be to attempt a circle-strafe maneuver, but that requires some more thought. A simple improvement might be to do this instead:
```C#
QueueChild(new Fly(self, target, target.CenterPosition));
QueueChild(new FlyForward(self, 1));
QueueChild(new Fly(self, target, exitRange, WDist.MaxValue, target.CenterPosition));
```
This is similar to the behaviour I described above, except it will try to face the target while doing it. (It kinda looks like something you might see in Battlestar Galactica actually. :smile: )